### PR TITLE
Refactor the SIG/meeting table

### PR DESCRIPTION
**Pull request by reyang**
Extracted the specification related SIGs to a separate table, added the TC sponsor column.
Fixed some small inconsistencies, better align with the OpenTelemetry glossary.

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -113,34 +113,41 @@ from 3 largely distinct timezones. The best way to report it and suggest an
 alternative is to file an issue on this repository or discuss it in
 SIG-specific GitHub discussions.
 
+### Specification SIGs
+
+Name|Meeting Time|Meeting Notes|Discussions|[Technical Committee](./community-members.md#technical-committee) Sponsors|
+----|------------|-------------|-----------|-----------|
+Specification: General|Every Tuesday at 08:00 PT|[Google Doc](https://docs.google.com/document/d/1pdvPeKjA8v8w_fGKAN68JjWBmVJtPCpqdi9IZrd6eEo/edit)|[Slack](https://cloud-native.slack.com/archives/C01N7PP1THC)
+Specification: Logs|Every Wednesday at 10:00 PT|[Google Doc](https://docs.google.com/document/d/1cX5fWXyWqVVzYHSFUymYUfWxUK5hT97gc23w595LmdM/edit#)|[Slack](https://cloud-native.slack.com/archives/C01N5UCHTEH)|[Tigran Najaryan](https://github.com/tigrannajaryan)
```
Comment (reyang) [inline README.md:121]: @tigrannajaryan FYI

**Review by tigrannajaryan — APPROVED** [review]

**Review by dyladan — COMMENTED** [review]
Much more discussion for JS happens in slack than in discussions. I would ay it should be kept in the discussions (with a link to the channel if possible)

**Comment by reyang** [comment]
> Much more discussion for JS happens in slack than in github discussions. I would ay it should be kept in the discussions column (with a link to the channel if possible)

This PR doesn't even touch the JS row, do you want it to be fixed here or in a separate PR?

**Comment by dyladan** [comment]
Hmm I was using the rendered view which shows the JS row changing. The rendered diff viewer must have been confused. It can be done in a separate PR. I'll handle that.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**